firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -51,10 +86,69 @@ private DefaultHeartBeatInfo(Context context) {
     return HeartBeat.NONE;
   }
 
+  @Override
+  public Task<List<HeartBeatResult>> getAndClearStoredHeartBeatInfo() {
+    TaskCompletionSource<List<HeartBeatResult>> taskCompletionSource = new TaskCompletionSource<>();
+    this.backgroundExecutor.execute(
+        () -> {
+          ArrayList<HeartBeatResult> heartBeatResults = new ArrayList<>();
+          boolean shouldSendGlobalHeartBeat = false;
+          List<SdkHeartBeatResult> sdkHeartBeatResults = storage.getStoredHeartBeats(true);
+          long lastGlobalHeartBeat = storage.getLastGlobalHeartBeat();
+          for (int i = 0; i < sdkHeartBeatResults.size(); i++) {
```

Comment:
if you don't need the indexes, pls use `for(Result result: sdkHeartBeatResults)`